File: app/src/main/java/gardezi/io/amiibofinder/dagger/MainActivityModule.java
package gardezi.io.amiibofinder.dagger;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import gardezi.io.amiibofinder.MainActivity;

@Module
public abstract class MainActivityModule {
    @ContributesAndroidInjector(modules = FragmentBuildersModule.class)
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/gardezi/io/amiibofinder/dagger/MainActivityModule.kt
package gardezi.io.amiibofinder.dagger

import dagger.Module
import dagger.android.ContributesAndroidInjector
import gardezi.io.amiibofinder.MainActivity

@Module
abstract class MainActivityModule {
    @ContributesAndroidInjector(modules = arrayOf(FragmentBuildersModule::class))
    internal abstract fun contributeMainActivity(): MainActivity
}